features/self-links.feature

```diff
@@ -0,0 +1,32 @@
+# language: nl
+
+Functionaliteit: JSON HAL self links worden opgenomen in de response
+  Bij self-link op het hoogste niveau van de response, dus _links.self.href is identiek aan de volledige url van het request, inclusief alle queryparameters. Een gebruiker moet hiermee de response aan de request kunnen koppelen. Deze kan dus niet templated zijn. Dit kan wel een relatieve link zijn.
+
+  Een self link in een embedded resource of in een collectie, dus een link in _embedded..[*]._links.self.href is een url die verwijst naar de betreffende resource. Deze link kan templated zijn.
+
+  Scenario: self link van een collectie
+    Als ingeschreven personen gezocht worden met "/ingeschrevenpersonen?naam__geslachtsnaam=groen&geboorte__datum=1983-05-26&inclusiefoverledenpersonen=true&fields=naam,geboorte,overlijden"
+    Dan eindigt attribuut "_links.self.href" op "/ingeschrevenpersonen?naam__geslachtsnaam=groen&geboorte__datum=1983-05-26&inclusiefoverledenpersonen=true&fields=naam,geboorte,overlijden"
+    En komt attribuut "templated" niet voor in "_links.self"
+
+  Scenario: self link van een resource in een collectie
+    Als ingeschreven personen gezocht worden met "/ingeschrevenpersonen?naam__geslachtsnaam=groen&geboorte__datum=1983-05-26&inclusiefoverledenpersonen=true&fields=naam,geboorte,overlijden"
+    Dan geldt voor elk van de gevonden _embedded.ingeschrevenpersonen dat attribuut _links.self.href de tekst "/ingeschrevenpersonen/" plus de waarde van attribuut burgerservicenummer in deze resource bevat
+    En geldt voor elk van de gevonden _embedded.ingeschrevenpersonen dat attribuut _links.self.href niet "naam__geslachtsnaam=groen" bevat
+    En geldt voor elk van de gevonden _embedded.ingeschrevenpersonen dat attribuut _links.self.href niet "geboorte__datum=1983-05-26" bevat
+    En geldt voor elk van de gevonden _embedded.ingeschrevenpersonen dat attribuut _links.self.href niet "inclusiefoverledenpersonen" bevat
+    En geldt voor elk van de gevonden _embedded.ingeschrevenpersonen dat attribuut _links.self.href niet "fields" bevat
+
+  Scenario: self link van een resource
+    Als ingeschreven persoon wordt geraadpleegd met "/ingeschrevenpersonen/999999023?fields=naam,geboorte,overlijden&expand=kinderen
+    Dan eindigt attribuut "_links.self.href" met "/ingeschrevenpersonen/999999023?fields=naam,geboorte,overlijden&expand=kinderen"
+
+  Scenario: self link van een embedded resource
+    Gegeven de persoon met burgerservicenummer 999999023 heeft een actuele partner
+    Als ingeschreven persoon wordt geraadpleegd met "/ingeschrevenpersonen/999999023?expand=partners.naam&fields=naam,geboorte,overlijden"
+    Dan bevat het attribuut "_embedded.partners[0]._links.self.href" de tekst "/ingeschrevenpersonen/999999023/partners/"
```

Comment:
Ik had hier de href naar de partner verwacht, dus als bsn van partner = 999999024, dan is deze href gelijk aan `/ingeschrevenpersonen/999999024`

Reply:
@MelvLee dit gaat om de self-link van de subresource partners. Deze wordt geïdentificeerd met een {id}, wat niet hetzelfde is als het burgerservicenummer van deze partner. Dit kan bijvoorbeeld een uuid zijn. Deze verwijst namelijk naar de subresource partner niet naar de ingeschrevenpersoon die de partner is.

Dus als het burgerservicenummer van de persoon 999999023 en van de partner 999999024 is, is _embedded.partners[0]._links.self.href bijvoorbeeld: "/ingeschrevenpersonen/999999023/partners/8a76s-ahij23-8a987".
Daarnaast is er dan een link _embedded.partners[0]._links.ingeschrevenPersoon.href met waarde "/ingeschrevenpersonen/999999024"

Reply:
Ik begrijp het weer. Maar wel verwarrend om Partner als voorbeeld te gebruiken omdat Partner, Kind en Ouder sub-resources zijn zonder identificatie. Misschien is een ander sub-resource die een identificatie heeft (zakelijkgerechtigde) als voorbeeld beter